xml-occurrence-parsers/src/test/java/org/gbif/xml/occurrence/parser/parsing/validators/UniquenessValidatorTest.java

```diff
@@ -0,0 +1,89 @@
+package org.gbif.xml.occurrence.parser.parsing.validators;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Stopwatch;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests the {@link UniquenessValidator}.
+ */
+public class UniquenessValidatorTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UniquenessValidatorTest.class);
+
+  @Test
+  public void givenUniqueIdsWhenMappedThenNoDuplicates() {
+    // increase it to test with a higher volume of data
+    int n = 1000;
+
+    try (UniquenessValidator validator = UniquenessValidator.getNewInstance()) {
+      for (int i = 0; i < n; i++) {
+        Assert.assertTrue(validator.isUnique(UUID.randomUUID().toString()));
+      }
+    }
+  }
+
+  @Test
+  public void givenDuplicatedIdsWhenMappedThenDuplicateFound() {
+    // increase it to test with a higher volume of data
+    int n = 10;
+    int duplicatesFound = 0;
+
+    try (UniquenessValidator validator = UniquenessValidator.getNewInstance()) {
+      // add duplicates
+      List<String> duplicates = new ArrayList<>(Arrays.asList("12.1", "1234.1", "111.1", "12.12"));
```

Comment:
new ArrayList<>(.. can be removed
and maybe something like this can help:
private static long countDuplicates(Collection<String> ids, UniquenessValidator validator) {
      return ids.stream().filter(validator::isUnique).count();
  }

or just use Arrays.asList("12.1", "1234.1", "111.1", "12.12")..filter(validator::isUnique).count(); ?


shouldn't we 'assert' after each partial test?